### ISSUE TITLE
False positive for UnnecessaryConversion

### ISSUE THREAD

**Issue by henrikbarium** [sksamuel/scapegoat#265]
Scapegoat version 1.4.1 will report the use of `toSet` on a mutable `Set` as an unnecessary conversion. It is not unnecessary as a mutable `Set` cannot be used as an immutable `Set`.

The example below would trigger this false positive.

```
val ms: scala.collection.mutable.Set[_] = mutable.Set()
val is: scala.collection.immutable.Set[_] = ms.toSet
                                               ^ not an unnecessary conversion
```

**Comment by mccartney** [comment]
It must've been introduced in #263

**Comment by mccartney** [comment]
I assume it has been fixed be #267 now. So let's close this.
